python/dgl/nn/pytorch/hetero.py

```diff
@@ -250,3 +251,137 @@ def get_aggregate_fn(agg):
         return _stack_agg_func
     else:
         return partial(_agg_func, fn=fn)
+
+class HeteroLinearLayer(nn.Module):
+    """Apply a linear transformation on the node features of a
+    heterogeneous graph and return a homogeneous graph representation.
+
+    The underlying implementation invokes dgl.to_homogeneous. Therefore,
+    the returned graph stores the following extra attributes:
+    
+      - ret_g.ndata[dgl.NID]: the original node IDs.
+      - ret_g.edata[dgl.EID]: the original edge IDs.
+      - ret_g.ndata[dgl.NTYPE]: the type ID of each node.
+      - ret_g.edata[dgl.ETYPE]: the type ID of each edge.
+
+    Parameters
+    ----------
+    hg : DGLGraph
+        A heterogeneous DGL graph.
+    out_size : int
+        Output feature size.
+    feat_name : str
+        Which feature to transform.
+
+    Examples
+    --------
+    >>> hg = dgl.heterograph({('user', 'rate', 'movie') : ...,
```

Comment:
Could you modify this example to make it runnable? Also I guess you need an extra empty line below "--------".